assignment/tool/src/java/org/sakaiproject/assignment/tool/AssignmentAction.java

```diff
@@ -1321,6 +1321,8 @@ public String buildMainPanelContext(VelocityPortlet portlet, Context context, Ru
         } catch (IdUnusedException iue) {
             log.warn(this + ":buildMainPanelContext: Site not found!" + iue.getMessage());
         }
+        User user = (User) state.getAttribute(STATE_USER);
+        context.put("isStudent", s.getUserRole(user.getId()).getId().equals("Student"));
```

Comment:
I think the correct permission should be checked instead of role.

Reply:
Exactly!

Reply:
Ok. I'll keep it in mind